src/main/scala/io/citrine/lolo/trees/splits/BoltzmannSplitter.scala

```diff
@@ -0,0 +1,239 @@
+package io.citrine.lolo.trees.splits
+
+import io.citrine.lolo.trees.impurity.VarianceCalculator
+import io.citrine.lolo.trees.splits.BoltzmannSplitter.SplitterResult
+
+import scala.collection.mutable
+import scala.util.Random
+
+/**
+  * Find a split for a regression problem
+  *
+  * The splits are picked with a probability that is related to the reduction in variance:
+  * P(split) ~ exp[ - {remaining variance} / ({temperature} * {total variance}) ]
+  * recalling that the "variance" here is weighted by the sample size (so its really the sum of the square difference
+  * from the mean of that side of the split).  This is akin to kinetic monte carlo and simulated annealing techniques.
```

Comment:
[trivial, nitpick] I might just say "metropolis-hastings" or plain "monte carlo". Kinetic MC highlights an orthogonal consideration (relative rate/event occurrence given fixed amount of elapsed time).